### PR TITLE
Feature/saas 10453 aws cloudfront  tls origins r

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: exports.js [hunk 1]
```diff
@@ -59,6 +59,8 @@ module.exports = {
         'enableOriginFailOver'          : require(__dirname + '/plugins/aws/cloudfront/enableOriginFailOver.js'),
         'cloudfrontGeoRestriction'      : require(__dirname + '/plugins/aws/cloudfront/cloudfrontGeoRestriction.js'),
         'compressObjectsAutomatically'  : require(__dirname + '/plugins/aws/cloudfront/compressObjectsAutomatically.js'),
+        'cloudfrontCustomOriginHttpsOnly' : require(__dirname + '/plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js'),
+
```
Comment (alphadev4) [inline exports.js:63]:
```suggestion
```

File: plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
+var helpers = require('../../../helpers/aws');
+
+module.exports = {
+    title: 'CloudFront Custom Origin HTTPS Only',
+    category: 'CloudFront',
+    domain: 'Content Delivery',
+    description: 'Ensures CloudFront Distribution Custom Origin is HTTPS Only.',
+    more_info: 'When you create a distribution, you specify the origin where CloudFront sends requests for the files. You can use several different kinds of origins with CloudFront.',
+    link: 'https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/DownloadDistS3AndCustomOrigins.html',
+    recommended_action: 'Modify cloudFront distribution and select Https only for custom origin.',
```
Comment (alphadev4) [inline plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js:10]:
```suggestion
    recommended_action: 'Modify CloudFront distribution and select Https only for custom origin.',
```

File: plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+ 
+        for (let distribution of listDistributions.data){
+            if (!distribution.ARN) continue;
+
+            if (!distribution.Origins || !distribution.Origins.Items || !distribution.Origins.Items.length){
+                helpers.addResult(results, 0, 'CloudFront distribution has no origins', 'global', distribution.ARN);
+            } else {
+                let items = distribution.Origins.Items;
+                for (let origin of items){
+                    if (!origin.CustomOriginConfig) continue;
```
Comment (AkhtarAmir) [inline plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js:43]:
```suggestion
                    if (!origin.CustomOriginConfig || !origin.CustomOriginConfig.OriginProtocolPolicy) continue;
```

File: plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+                    let originProtocolPolicy = origin.CustomOriginConfig.OriginProtocolPolicy.toLowerCase();
+
+                    if (originProtocolPolicy == 'http-only' || originProtocolPolicy == 'match-viewer'){
```
Comment (AkhtarAmir) [inline plugins/aws/cloudfront/cloudfrontCustomOriginHttpsOnly.js:47]: do this outside loop

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

**Pull request merged**